bpfd-operator/README.md

```diff
@@ -1,80 +1,179 @@
 ```
 
-### Uninstall CRDs
-To delete the CRDs from the cluster:
+### API
+
+#### EbpfProgramConfig
+
+The `EbpfProgramConfig` crd is the bpfd K8s API object most relevant to users and can be used to understand clusterwide state for an ebpf program. It's designed to express how, and where bpf programs are to be deployed within a kubernetes cluster.  An example ebpfProgramConfig which loads a basic `xdp-pass` program to all nodes can be seen below:
+
+**NOTE: Currently the ebpfprogram's bytecode section-name MUST match the `spec.name` field in the ebpfProgramConfig Object.**
+
+```yaml
+apiVersion: bpfd.io/v1alpha1
+kind: EbpfProgramConfig
+metadata:
+  labels:
+    app.kubernetes.io/name: ebpfprogramconfig
+  name: xdp-pass-all-nodes
+spec:
+  ## Must correspond to image section name
+  name: pass
+  type: XDP
+  # Select all nodes
+  nodeselector: {}
+  priority: 0
+  attachpoint: 
+    interface: eth0
+  bytecode:
+    imageurl: quay.io/bpfd/bytecode:xdp_pass
+```
+
+### EbpfProgram
+
+The `EbpfProgram` crd is used internally by the bpfd-deployment to keep track of per node bpfd state such as program UUIDs and map pin points, and to report node specific errors back to the user. K8s users/controllers are only allowed to view these objects, NOT create or edit them.  Below is an example ebpfProgram Object which was automatically generated in response to the above ebpfProgramConfig Object.
+
+```yaml
+apiVersion: bpfd.io/v1alpha1
+  kind: EbpfProgram
+  metadata:
+    creationTimestamp: "2022-12-07T22:41:29Z"
+    finalizers:
+    - bpfd.io.agent/finalizer
+    generation: 2
+    labels:
+      owningConfig: xdp-pass-all-nodes
+    name: xdp-pass-all-nodes-bpfd-deployment-worker2
+    ownerReferences:
+    - apiVersion: bpfd.io/v1alpha1
+      blockOwnerDeletion: true
+      controller: true
+      kind: EbpfProgramConfig
+      name: xdp-pass-all-nodes
+      uid: 6e3f5851-97b1-4772-906b-3ac69c6a4057
+    resourceVersion: "1506"
+    uid: 384d3d5c-e62b-4be3-9bf0-c6cf0e315acf
+  spec:
+    programs:
+      bdeac6d3-4128-464e-9161-6010684eca27:
+        attachpoint:
+          interface: eth0
+        maps: {}
+  status:
+    conditions:
+    - lastTransitionTime: "2022-12-07T22:41:30Z"
+      message: Successfully loaded ebpfProgram
+      reason: bpfdLoaded
+      status: "True"
+      type: Loaded
+```
+
+Applications wishing to use bpfd to deploy/manage their bpf programs in kubernetes will make use of this
+object to find references to the bpfMap pin points (`spec.maps`) in order to configure their bpf programs.
+
+### Controllers
+
+The Bpfd-Operator performs a few major functions and houses two major controllers the `bpfd-agent` and `bpfd-operator`.
+
+#### bpf-agent
+
+The bpfd-agent controller is deployed alongside bpfd in a daemonset.  It's main purpose to to watch user intent (in ebpfProgramConfig Objects) and communicate with
+bpfd via an mTLS secured connection in order to translate the cluster-wide user-inetent to per node state.
```

Comment:
```suggestion
bpfd via a mTLS secured connection in order to translate the cluster-wide user-intent to per node state.
```
Not sure if it's `a` or `an`, I think `a`.

Reply:
Done